extensions/parquet/base/src/main/java/io/deephaven/parquet/base/util/SeekableChannelsProvider.java

```diff
@@ -3,22 +3,70 @@
  */
 package io.deephaven.parquet.base.util;
 
+import io.deephaven.util.SafeCloseable;
+import io.deephaven.util.annotations.FinalDefault;
 import org.jetbrains.annotations.NotNull;
 
+import java.io.File;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.channels.SeekableByteChannel;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
-public interface SeekableChannelsProvider {
+public interface SeekableChannelsProvider extends SafeCloseable {
 
-    default SeekableByteChannel getReadChannel(@NotNull final String path) throws IOException {
-        return getReadChannel(Paths.get(path));
+    /**
+     * Take the file source path or URI and convert it to a URI object.
+     *
+     * @param source The file source path or URI
+     * @return The URI object
+     */
+    static URI convertToURI(final String source) {
+        final URI uri;
+        try {
+            uri = new URI(source);
+        } catch (final URISyntaxException e) {
+            // If the URI is invalid, assume it's a file path
+            return new File(source).toURI();
+        }
+        if (uri.getScheme() == null) {
+            // Need to convert to a "file" URI
+            return new File(source).toURI();
+        }
+        return uri;
     }
 
-    SeekableByteChannel getReadChannel(@NotNull Path path) throws IOException;
+    /**
+     * Create a new {@link SeekableChannelContext} object for creating read and write channels via this provider.
+     */
+    SeekableChannelContext makeContext();
 
-    default SeekableByteChannel getWriteChannel(@NotNull final String path, final boolean append) throws IOException {
+    /**
+     * Check if the given context is compatible with this provider. Useful to test if we can use provided
+     * {@code context} object for creating channels with this provider.
+     */
+    boolean isCompatibleWith(@NotNull SeekableChannelContext channelContext);
+
+    interface ContextHolder {
```

Comment:
Seems to me like this API is more for `CachedChannelProvider` than `SeekableChannelsProvider`.

Reply:
You are right. None of the `SeekableChannelsProvider` even use the context object passed to the channel in `getReadChannel` other than `CachedChannelProvider`.
Do you think we need to do something about this?